lib/hexpm/repository/package.ex

```diff
@@ -115,6 +115,20 @@ defmodule Hexpm.Repository.Package do
     end
   end
 
+  defp search_param("depends", search, query) do
+    dependencies = from(rq in Requirement,
+      join: p in assoc(rq, :dependency),
+      join: rl in assoc(rq, :release),
+      select: rl.package_id,
+      distinct: rl.package_id,
+      where: p.name == ^search,
+    )
+
+    from(p in query,
+      join: d in subquery(dependencies),
```

Comment:
You can do it without subqueries:

```elixir
from(p1 in query,
     join: rel in assoc(p1, :releases),
     join: req in assoc(rel, :requirements),
     join: p2 in assoc(p2, :dependency),
     where: p2.name == ^name,
     distinct: p.id)
```

This may not be very performant, we will have to measure/EXPLAIN on production data. It doesn't have to be very quick and we are very read heavy so we can even have a materialized view for this that we update on every package publish.